cdap-app-fabric/src/main/java/io/cdap/cdap/internal/accelerator/AcceleratorManager.java

```diff
@@ -0,0 +1,138 @@
+    this.metadataAdmin = metadataAdmin;
+    this.cConf = cConf;
+  }
+
+  /**
+   * Returns the list of applications that are having metadata tagged with the accelerator
+   *
+   * @param namespace
+   * @param accelerator
+   * @return
+   * @throws Exception
+   */
+  public Set<ApplicationId> getAppsForAccelerator(NamespaceId namespace,
+                                                  String accelerator) throws Exception {
+    String acceleratorTag = String.format(ACCELERATOR_TAG, accelerator);
+    SearchRequest searchRequest = SearchRequest.of(acceleratorTag).addNamespace(namespace.getNamespace())
+      .setScope(MetadataScope.SYSTEM).build();
+    SearchResponse searchResponse = metadataAdmin.search(searchRequest);
+    List<MetadataRecord> results = searchResponse.getResults();
+    Set<ApplicationId> applicationIds = results.stream()
+      .map(MetadataRecord::getEntity)
+      .filter(this::isApplicationType)
+      .map(this::getApplicationId)
+      .collect(Collectors.toSet());
+    return applicationIds;
+  }
+
+  /**
+   * Returns boolean indicating whether application is disabled due to a disabled accelerator
+   *
+   * @param namespace
+   * @param applicationName
+   * @return
+   * @throws Exception
+   */
+  public boolean isApplicationDisabled(String namespace, String applicationName) throws Exception {
+    String applicationQuery = String.format(APPLICATION_TAG, applicationName);
+    SearchResponse searchResponse = metadataAdmin
+      .search(SearchRequest.of(applicationQuery).addNamespace(namespace).setScope(MetadataScope.SYSTEM).build());
+    Set<String> acceleratorTagsForApp = searchResponse.getResults().stream()
+      .filter(this::hasAcceleratorTagValue)
+      .map(this::getAcceleratorTagValue)
+      .collect(Collectors.toSet());
+    for (String acceleratorTagForApp : acceleratorTagsForApp) {
+      if (isAcceleratorDisabled(acceleratorTagForApp)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private String getAcceleratorTagValue(MetadataRecord metadataRecord) {
+    return metadataRecord.getMetadata().getProperties().get(new ScopedName(MetadataScope.SYSTEM, ACCELERATOR_KEY));
+  }
+
+  private boolean hasAcceleratorTagValue(MetadataRecord metadataRecord) {
+    return metadataRecord.getMetadata().getProperties()
+      .get(new ScopedName(MetadataScope.SYSTEM, ACCELERATOR_KEY)) != null;
+  }
+
+  private boolean isApplicationType(MetadataEntity metadataEntity) {
+    return MetadataEntity.APPLICATION.equals(metadataEntity.getType());
+  }
+
+  private ApplicationId getApplicationId(MetadataEntity metadataEntity) {
+    return new ApplicationId(metadataEntity.getValue(MetadataEntity.NAMESPACE),
+                             metadataEntity.getValue(MetadataEntity.APPLICATION),
+                             metadataEntity.getValue(MetadataEntity.VERSION));
+  }
+
+  private boolean isAcceleratorDisabled(String accelerator) {
```

Comment:
Annotate the parameter with `@Nullable`.

Reply:
Done